### ISSUE TITLE
Healthcheck fails when the API is excluded

### ISSUE THREAD

**Issue by kristof-mattei** [immich-app/immich#10157]
### The bug

Split workers, as per the v1.106.1, have a broken healthcheck on the new 'microservices' container:

The healthcheck should take into account that a container might not be running an API at 3001.

This becomes problematic in environment where there is some sort of autoheal service that restarts containers if they are unhealthy. 

Source: https://github.com/immich-app/immich/blob/69b948f3d09aeb6d795fe948471d38ef0d57d9bf/server/src/utils/healthcheck.ts#L8-L10

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v1.106.1

### Version of Immich Mobile App

vIrrelevant

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
- name: Create immich-server container
      community.general.docker_container:
        container_default_behavior: "no_defaults"
        name: immich-server
        image: ghcr.io/immich-app/immich-server:release
        restart_policy: always
        env:
          ...
          IMMICH_WORKERS_INCLUDE: "api"
          ...
        mounts:
          - ...
        networks:
          - name: bridge_ipv4
        labels:
          ...

    - name: Create immich-microservices container
      community.general.docker_container:
        container_default_behavior: "no_defaults"
        name: immich-microservices
        image: ghcr.io/immich-app/immich-server:release
        restart_policy: always
        env:
          ...
          IMMICH_WORKERS_EXCLUDE: "api"
          ...
        devices:
          - /dev/dri:/dev/dri
        mounts:
          - ...
        networks:
          - name: bridge_ipv4
```

### Your .env content

```Shell
Irrelevant
```

### Reproduction steps

```bash
1. Deploy separate containers for API and worker
2. Observe worker's healthcheck failing
```

### Relevant log output

```shell
> immich@1.106.1 healthcheck > node ./dist/utils/healthcheck.js TypeError: fetch failed at node:internal/deps/undici/undici:12502:13 at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async main (/usr/src/app/dist/utils/healthcheck.js:9:26) { [cause]: Error: connect ECONNREFUSED 127.0.0.1:3001 at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1606:16) { errno: -111, code: 'ECONNREFUSED', syscall: 'connect', address: '127.0.0.1', port: 3001 } }
```

**Comment by zackpollard** [comment]
Good shout, totally didn't consider that case, will fix this up tomorrow and get another hotfix release out :smiley:

**Comment by kristof-mattei** [comment]
Thanks!

**Comment by alextran1502** [comment]
This is fixed, correct? @zackpollard

**Comment by zackpollard** [comment]
> This is fixed, correct? @zackpollard 

Yep. Forgot to close it, should be fixed in 1.106.3
